### DISCUSSION TITLE
Using latest gstreamer

### DISCUSSION THREAD

**Discussion by goetzdd** [OE4T/tegra-demo-distro#143] [General]
I was working with the master branch (from several weeks (months?) ago) and previously I had just removed 'require conf/include/gstreamer-1.14.conf' from ../tegra-demo-distro/layers/meta-tegrademo/conf/distro/tegrademo.conf and had no issues bitbaking my images. 

I realized I should maybe be working from hardknott instead of bleeding edge (though I hadn't git pull so not up to date) so switched branch to track hardknott and also realized instead of modifying tegrademo.conf it appears I can just set the version in my conf overriding GST_VERSION (?= "1.18%) in my conf. But while this appears to then select 1.18 I now have warnings on bitbake:

WARNING: preferred version 1.18% of gstreamer1.0-plugins-bad not available (for item libgstvideo-1.0)
WARNING: versions of gstreamer1.0-plugins-bad available: 1.14.4
WARNING: preferred version 1.18% of gstreamer1.0-plugins-good not available (for item libgstvideo-1.0)
WARNING: versions of gstreamer1.0-plugins-good available: 1.14.4
WARNING: preferred version 1.18% of gstreamer1.0-plugins-ugly not available (for item libgstvideo-1.0)
WARNING: versions of gstreamer1.0-plugins-ugly available: 1.14.4
WARNING: preferred version 1.18% of gstreamer1.0-rtsp-server not available (for item libgstvideo-1.0)
WARNING: versions of gstreamer1.0-rtsp-server available: 1.14.4
NOTE: Multiple providers are available for runtime libgstvideo-1.0 (gstreamer1.0-plugins-base, gstreamer1.0-plugins-bad, gstreamer1.0-plugins-good, gstreamer1.0-plugins-ugly, gstreamer1.0-rtsp-server)
Consider defining a PREFERRED_RPROVIDER entry to match libgstvideo-1.0

I've been using gstreamer 1.18 with the 1.14 nvidia provided plugins, but specifically the nvvidconv and nvoverlaysink plugins so not terribly concerned about that mismatch. I figure maybe I need to follow the "Consider..." comment but when I go looking for PREFERRED_RPROVIDER I don't even find that defined in the yocto manual (https://docs.yoctoproject.org/singleindex.html).

Then I go poking about and see yocto has made a revision change as well so confusion...

So I have to minimum be at hardknott for gstreamer 1.18 it appears (dunfell has 1.16.3)..

So I suppose i'm looking to see if the warnings above will be fatal, but I suppose I'll find out when I try the image as well...

**Comment by madisongh** [comment]
You can ignore the warnings.  They're caused by [this change](https://git.yoctoproject.org/cgit/cgit.cgi/poky/commit/meta/recipes-multimedia/gstreamer?id=768d6210677953cea4e1db5a5fc6be4c6598f36a) to the gstreamer recipes, and they're noise caused by the extra dynamic package names being generated. `libgstvideo-1.0` *isn't* actually provided by any of the packages mentioned - it's only actually provided by `gstreamer1.0-plugins-base`.

I saw the same problem (in the other direction) when this change got ported over to the 1.14 recipes we have in meta-tegra/contrib... I just got it [fixed there](https://github.com/OE4T/meta-tegra/commit/a5f85b06b30f77afa340f96b630b66153442403c#diff-2d16c7e5e0fa00494183652f0c06e3bc7d67224ad9ffd737712589e314d8da91) a couple of weeks ago, and haven't had a chance to submit a similar patch for the upstream recipes.

**Reply by goetzdd** [reply]
Thanks!
